native/native_image.go

```diff
@@ -225,3 +210,42 @@ func (n NativeImage) ProcessArguments(layer libcnb.Layer) ([]string, string, err
 func (NativeImage) Name() string {
 	return "native-image"
 }
+
+// copy the main file & any `*.so` files also in the layer to the application path
+func copyFilesFromLayer(layerPath string, execName string, appPath string) error {
+	files, err := os.ReadDir(layerPath)
+	if err != nil {
+		return fmt.Errorf("unable to list files on layer %s\n%w", layerPath, err)
+	}
+
+	for _, file := range files {
+		if file.Type().IsRegular() && (file.Name() == execName) {
+			src := filepath.Join(layerPath, file.Name())
+			dst := filepath.Join(appPath, file.Name())
+
+			if err := copyFile(src, dst); err != nil {
+				return fmt.Errorf("unable to copy %s to %s\n%w", src, dst, err)
+			}
+		}
+		if file.Type().IsRegular() && (strings.HasSuffix(file.Name(), ".so")) {
+			src := filepath.Join(layerPath, file.Name())
+			dst := filepath.Join(appPath, file.Name())
+
+			if err := copyFile(src, dst); err != nil {
+				return fmt.Errorf("unable to copy %s to %s\n%w", src, dst, err)
+			}
+		}
+	}
+
+	return nil
+}
+
+func copyFile(src string, dst string) error {
+	in, err := os.Open(src)
+	if err != nil {
+		return fmt.Errorf("unable to open %s\n%w", src, err)
+	}
+	defer in.Close()
+
+	return sherpa.CopyFile(in, dst)
```

Comment:
there's no signature of `sherpa.CopyFile` that does the Open + error handling + defer close?

Reply:
Unfortunately no. I just added a story for that: https://github.com/paketo-buildpacks/libpak/issues/354